.github/workflows/llvm-build.yml

```diff
@@ -0,0 +1,194 @@
+name: LLVM Build
+
+on:
+  push:
+    branches:
+      - llvm-head
+    paths:
+      - llvm-hash.txt
+  workflow_dispatch:
+
+env:
+  SCCACHE_DIR: ${{ github.workspace }}/sccache
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+
+  build:
+
+    strategy:
+      fail-fast: true
+
+      matrix:
+        platform: [
+          ubuntu-20.04-x64,
+          ubuntu-22.04-x64,
+          centos-7-x64,
+          macos-x64,
+          macos-arm64
+        ]
+
+        include:
+          # Specify OS versions
+          - platform: ubuntu-20.04-x64
+            host-os: ubuntu-20.04
+            target-os: ubuntu
+            arch: x64
+          - platform: ubuntu-22.04-x64
+            host-os: ubuntu-22.04
+            target-os: ubuntu
+            arch: x64
+          - platform: centos-7-x64
+            host-os: ubuntu-22.04
+            target-os: centos
+            arch: x64
+          - platform: macos-x64
+            host-os: macos-12
+            target-os: macos
+            arch: x64
+          - platform: macos-arm64
+            host-os: macos-12
+            target-os: macos
+            arch: arm64
+
+    runs-on: ${{ matrix.host-os }}
+
+    if: github.ref == 'refs/heads/llvm-head'
```

Comment:
Do we need this? If someone wanted to manually build LLVM from a different branch for some reason that seems fine to support.

Reply:
The llvm-hash.txt file exists only in the llvm-head branch, so I don't think we would want to run this workflow in a different branch, would we?

Reply:
I'm imaging someone makes a branch off llvm-head to prototype the update, so it would have llvm-hash.txt. If someone tried ina branch that didn't have it the workflow would just fail.

Reply:
Sounds good.  The only branch restriction now is that the workflow is automatically triggered for the llvm-head branch only (and when the llvm-hash.txt file is modified).  The workflow can be invoked manually for any branch now.